llvm/test/CodeGen/X86/pr50907.ll

```diff
@@ -2,7 +2,7 @@
 ; RUN: not llc -o /dev/null %s 2>&1 | FileCheck %s
 target triple = "x86_64-unknown-linux-gnu"
 
-; CHECK: error: couldn't allocate input reg for constraint 'r'
+; CHECK: error: couldn't allocate input reg for constraint 'r': couldn't allocate for type v8i16
```

Comment:
Maybe we can be more specific?  Something like "expected an integer, found v8i16"?

Reply:
This is hard. To be precise and informative we need to name all the vector combinations that can fit in 128 bits in case of 'x' constraint and even naming is not simple, as we need to understand whether `v8f16` is available or not to include it into the message.  Or do you mean `r` constraint in particular?